ui/perfherder/health/HealthView.jsx

```diff
@@ -0,0 +1,176 @@
+      getData('api/machineplatforms/').then(({ data, failureStatus }) => {
+        if (failureStatus) {
+          this.setState({ failureMessages: [data, ...failureMessages] });
+        } else {
+          updateAppState({ platforms: data });
+          this.createObjectsMap(data, 'platformsMap', 'platform');
+        }
+      });
+    }
+  };
+
+  getTestSuiteHealthData = async params => {
+    const { failureMessages } = this.state;
+
+    const { data, failureStatus } = await getData(
+      createApiUrl('performance/validity-dashboard/', params),
+    );
+    if (failureStatus) {
+      return { failureMessages: [data, ...failureMessages] };
+    }
+    return { results: data };
+  };
+
+  createObjectsMap = (objects, state, propertyName) => {
+    const objectsMap = objects.reduce((result, currentObject) => {
+      result[currentObject.id] = currentObject[propertyName];
+      return result;
+    }, {});
+
+    this.setState({ [state]: objectsMap });
+  };
+
+  updateFramework = selection => {
+    const { updateParams } = this.props.validated;
+    const { frameworks } = this.props;
+    const framework = frameworks.find(item => item.name === selection);
+
+    updateParams({ framework: framework.id });
+    this.setState({ framework }, () => this.getHealthData());
+  };
+
+  render() {
+    const { frameworks } = this.props;
+    const {
+      framework,
+      results,
+      loading,
+      failureMessages,
+      projectsMap,
+      platformsMap,
+    } = this.state;
+
+    const frameworkNames =
+      frameworks && frameworks.length ? frameworks.map(item => item.name) : [];
+
+    const healthDropdows = [
+      {
+        options: frameworkNames,
+        selectedItem: framework.name,
+        updateData: this.updateFramework,
+      },
+    ];
+
+    return (
+      <ErrorBoundary
+        errorClasses={errorMessageClass}
+        message={genericErrorMessage}
+      >
+        <Container fluid className="max-width-default">
+          {loading && !failureMessages.length && <LoadingSpinner />}
+          <Row className="justify-content-center">
+            <Col sm="8" className="text-center">
+              {failureMessages.length !== 0 && (
+                <ErrorMessages errorMessages={failureMessages} />
+              )}
+            </Col>
+          </Row>
+          <Row>
+            <Col sm="12" className="text-center pb-1">
+              <h1>Perfherder Health</h1>
+            </Col>
+          </Row>
+          <HealthTableControls
+            healthResults={results}
+            dropdownOptions={healthDropdows}
+            projectsMap={projectsMap}
+            platformsMap={platformsMap}
+          />
+        </Container>
+      </ErrorBoundary>
+    );
+  }
+}
+
+HeathView.propTypes = {
+  location: PropTypes.shape({}),
+  validated: PropTypes.shape({
+    projects: PropTypes.arrayOf(PropTypes.shape({})),
+    frameworks: PropTypes.arrayOf(PropTypes.shape({})),
```

Comment:
I think these two props are only passed through `withValidation` and are not actually part of the `validated` props.